server/src/audit/audit.decorator.ts

```diff
@@ -0,0 +1,12 @@
+import { applyDecorators, SetMetadata, UseGuards } from '@nestjs/common';
+
+import { JwtAuthGuard } from 'src/auth/jwt-auth.guard';
+import { AuditGuard } from './audit.guard';
+
+export const AUDITABLE_KEY = 'endpointAuditable';
+
+export const Auditable = () =>
+  applyDecorators(
+    SetMetadata(AUDITABLE_KEY, true),
+    UseGuards(JwtAuthGuard, AuditGuard),
+  );
```

Comment:
You can just add `AuditGuard` to the existing `Auth` decorator, since it is not possible to run Audit without the previous authorization steps anyways. The `Auditable` decorator can just be a simple function that sets the meta data:

```ts
export const Auditable = () => SetMetadata(AUDITABLE_KEY, true);

// in your endpoint

class SomEndpoint {
  @Post()
  @Auth(/** stff*/)
  @Auditable()
  thisIsMyEndpoint() {}
}
```